### PR TITLE
Add docs and make it work seamlessly on Debian 10

### PR DESCRIPTION
**Pull request by vladionescu**
- Add a Quick Start section to README, including a sample playbook, resolving #2 
- Change default system Python version from 3.6 to 3.7, resolving #1 
  Python 3.7 comes default on the latest Debian, and 3.6 is not available in a package anymore. Pretalx runs well under 3.7.
- Standardize Redis and Nginx vars. When set to `false` that component is disabled, otherwise the var should point to a location, which the var name hints at.
- Add a var to conditionally enable TLS (on by default).
- Add an HTTP->HTTPS redirect in Nginx when TLS is on.
- Change Nginx TLS cert path to where certbot drops certs.
  https://galaxy.ansible.com/geerlingguy/certbot is the most downloaded result for LetsEncrypt in Galaxy, and it uses certbot. This change is compatible with that, in anticipation of people taking the route of least friction when setting up their site.
- Remove `$request_basename` from Nginx config since that variable doesn't exist in the latest Nginx and thus breaks Ansible runs (by preventing Nginx from starting up).

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -27,8 +27,10 @@ pretalx_mail_password: password
 pretalx_mail_tls: "False"
 pretalx_mail_ssl: "True"
 
-pretalx_nginx: false
-pretalx_nginx_path: false
+pretalx_nginx_conf_dir: false
```
Comment (rixx) [inline defaults/main.yml]: Could you please reverse the renaming of existing configuration variables? This would break any existing use of this role for users, and since ansible galaxy does not have any proper dependency management, I'd rather not introduce the risk.

File: templates/nginx.conf [hunk 2]
```diff
@@ -4,14 +4,32 @@ upstream pretalx_{{ pretalx_instance_identifier }}_server {
...
+    server_name {{ pretalx_domain }}{% if pretalx_alternate_domains %} {{ pretalx_alternate_domains }}{% endif %};
+    return 301 https://{{ pretalx_domain }}$request_uri;
+}
+{% endif %}
+
+server {
+    {% if pretalx_tls %}
     listen 443 ssl;
     listen [::]:443 ssl;
+    {% else %}
```
Comment (rixx) [inline templates/nginx.conf]: Hmm, I'm not sure about this. pretalx should not run without HTTPS, and if people want to run it without, I'd recommend they use their own nginx configuration. I don't really want to make it easier to do than it has to be.

File: templates/nginx.conf [hunk 3]
```diff
@@ -4,14 +4,32 @@ upstream pretalx_{{ pretalx_instance_identifier }}_server {
...
+    listen [::]:80;
+    {% endif %}
+
     server_name {{ pretalx_domain }}{% if pretalx_alternate_domains %} {{ pretalx_alternate_domains }}{% endif %};
 
+    {% if pretalx_tls %}
     ssl on;
-    ssl_certificate /etc/ssl/letsencrypt/certs/{{ pretalx_domain }}/fullchain.pem;
-    ssl_certificate_key /etc/ssl/letsencrypt/certs/{{ pretalx_domain }}/privkey.pem;
+    ssl_certificate /etc/letsencrypt/archive/{{ pretalx_domain }}/fullchain1.pem;
```
Comment (rixx) [inline templates/nginx.conf]: Please revert this change, as it will break the behaviour of this role for existing users.

File: templates/nginx.conf [hunk 4]
```diff
@@ -40,7 +58,7 @@ server {
 
     location /media/ {
         alias {{ pretalx_media_dir }};
-        add_header Content-Disposition 'attachment; filename="$request_basename"';
+        add_header Content-Disposition 'attachment';
```
Comment (rixx) [inline templates/nginx.conf]: Could you explain the reason for this change, please?

File: defaults/main.yml (hunk 1 above)
Comment (vladionescu) [reply defaults/main.yml]: Good point, will do!

File: templates/nginx.conf (hunk 4 above)
Comment (vladionescu) [reply templates/nginx.conf]: Nginx fails to load configs with this line in it, saying that `$request_basename` isn't valid. This breaks the Ansible run, too, since it can't start Nginx.

**Comment by vladionescu** [comment]
Made the requested changes, check it out and let me know if it's acceptable. Thanks!

File: templates/nginx.conf [hunk 5]
```diff
@@ -4,9 +4,18 @@ upstream pretalx_{{ pretalx_instance_identifier }}_server {
 
 proxy_cache_path /tmp/nginx-pretalx-{{ pretalx_instance_identifier }} levels=1:2 keys_zone=pretalx_static_{{ pretalx_instance_identifier }}:10m inactive=60m max_size=250m;
 
+server {
```
Comment (rixx) [inline templates/nginx.conf]: Since this is a marked change in behaviour, it should maybe be hidden behind a flag.

File: templates/nginx.conf [hunk 6]
```diff
@@ -4,9 +4,18 @@ upstream pretalx_{{ pretalx_instance_identifier }}_server {
...
+    listen 80;
+    listen [::]:80;
+
+    server_name {{ pretalx_domain }}{% if pretalx_alternate_domains %} {{ pretalx_alternate_domains }}{% endif %};
+    return 301 https://{{ pretalx_domain }}$request_uri;
```
Comment (rixx) [inline templates/nginx.conf]: I'm not sure this is strictly correct – this will always redirect to the main domain on HTTPS, instead of the current request hostname.

File: README.rst [hunk 7]
```diff
@@ -19,3 +15,77 @@ to the `pretalx configuration documentation`_ for further details.
...
 .. _pretalx configuration documentation: https://docs.pretalx.org/en/latest/administrator/configure.html
+
+Quick Start
+-----------
+
+You will need an Ansible playbook that uses this role, a database for Pretalx to connect to, and Nginx to front the Python server. 
+
+You can configure variables in that playbook, overriding anything in the defaults_.
+
+The following steps were tested on Debian 10:
```
Comment (rixx) [inline README.rst]: Could you also add the Ansible and pretalx versions you used for testing, please? Ansible versions tend to have horrible backwards compatibility, and if this guide is prominent in the README (which is great btw, thank you!), it should be dated to avoid future frustration.

Comment (vladionescu) [reply README.rst]: I did not know that about Ansible! Added.

**Comment by vladionescu** [comment]
No problem, I appreciate the discussion, I want to make sure my contributions are going to be good.

**Comment by rixx** [comment]
I'll switch the `force_https` flag to default to `false` to keep in line with current behaviour after the merge (just a heads-up, since I assume you're using it). Thank you a lot for the PR, and your patience!
